rig/machine_control/machine_controller.py

```diff
@@ -0,0 +1,382 @@
+import collections
+import pkg_resources
+import socket
+import struct
+
+from .consts import SCPCommands, DataType
+from . import boot, consts
+from .scp_connection import SCPConnection
+from rig.utils.contexts import ContextMixin, Required
+
+
+class MachineController(ContextMixin):
+    """TODO
+
+    Attributes
+    ----------
+    """
+    def __init__(self, initial_host, n_tries=5, timeout=0.5, app_id=None,
+                 boot_data=None, struct_data=None):
+        """Create a new controller for a SpiNNaker machine.
+
+        Parameters
+        ----------
+        initial_host : string
```

Comment:
"Hostname or IP address of the SpiNNaker chip to connect to. If the board has not yet been booted, this will become chip (0, 0)."